### ISSUE TITLE
BUG: Code not running after installingPyGEOS

### ISSUE THREAD

**Issue by antonindanalet** [geopandas/geopandas#1652]
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of geopandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of geopandas.

---

#### Code Sample, a copy-pastable example

```
world = geopandas.read_file(geopandas.datasets.get_path('naturalearth_lowres'))
world.head()
```

#### Problem description

I have installed PyGEOS because my process is slow and I read this: [Using the optional PyGEOS dependency](https://geopandas.org/install.html#using-the-optional-pygeos-dependency). For a code that was working without PyGEOS, I get now a "UserWarning: The Shapely GEOS version (3.4.3-CAPI-1.8.3 r4285) is incompatible with the GEOS version PyGEOS was compiled with (3.8.1-CAPI-1.13.3). Conversions between both will be slow.
  shapely_geos_version, geos_capi_version_string" and "Process finished with exit code -1073741819 (0xC0000005)".

#### Expected Output

The code should simply run faster than without PyGEOS, ideally. The output should be:

```   GRNR  ...                                           geometry
0     1  ...  MULTIPOLYGON (((2546732.494 1193708.955, 25465...
1     2  ...  MULTIPOLYGON (((2549140.721 1178014.176, 25493...
2     3  ...  MULTIPOLYGON (((2595270.119 1251478.978, 25942...
3     4  ...  POLYGON ((2692430.313 1281173.385, 2692994.572...
4     5  ...  MULTIPOLYGON (((2686913.929 1272389.921, 26868...
5     6  ...  POLYGON ((2662038.335 1237690.711, 2662272.506...
6     7  ...  POLYGON ((2716299.544 1165466.590, 2716755.976...
```

#### Output of ``geopandas.show_versions()``

**Without PyGEOS:**

```
SYSTEM INFO
-----------
python     : 3.6.1 (v3.6.1:69c0db5, Mar 21 2017, 18:41:36) [MSC v.1900 64 bit (AMD64)]
executable : C:\...\python.exe
machine    : Windows-10-10.0.17763-SP0

GEOS, GDAL, PROJ INFO
---------------------
GEOS       : None
GEOS lib   : None
GDAL       : 3.1.2
GDAL data dir: None
PROJ       : 7.0.1
PROJ data dir: C:\...\lib\site-packages\pyproj\proj_dir\share\proj

PYTHON DEPENDENCIES
-------------------
geopandas  : 0.8.1
pandas     : 1.0.1
fiona      : 1.8.16
numpy      : 1.18.2
shapely    : 1.7.1
rtree      : 0.9.4
pyproj     : 2.6.1.post1
matplotlib : 3.1.1
mapclassify: None
geopy      : None
psycopg2   : None
geoalchemy2: None
pyarrow    : None
None
```

**With PyGEOS:**

```
SYSTEM INFO
-----------
python     : 3.6.1 (v3.6.1:69c0db5, Mar 21 2017, 18:41:36) [MSC v.1900 64 bit (AMD64)]
executable : C:\...\python.exe
machine    : Windows-10-10.0.17763-SP0

GEOS, GDAL, PROJ INFO
---------------------
GEOS       : None
GEOS lib   : None
GDAL       : 3.1.2
GDAL data dir: None
PROJ       : 7.0.1
PROJ data dir: C:\...\lib\site-packages\pyproj\proj_dir\share\proj

PYTHON DEPENDENCIES
-------------------
geopandas  : 0.8.1
pandas     : 1.1.1
fiona      : 1.8.16
numpy      : 1.19.1
shapely    : 1.7.1
rtree      : 0.9.4
pyproj     : 2.6.1.post1
matplotlib : None
mapclassify: None
geopy      : None
psycopg2   : None
geoalchemy2: None
pyarrow    : None
None
```

</details>

**Comment by jorisvandenbossche** [comment]
Hi @antonindanalet, first, thanks a lot for trying out PyGEOS!

The warning can certainly happen (like it says, the GEOS version used by Shapely and GEOS is different), but the crash certainly should *not* happen, even when different GEOS versions are used.

I am a bit surprised that Shapely is built against such an old GEOS version, though. The warning indicates it is using GEOS 3.4.3 (we don't even test that in pygeos, and also not in Shapely, it seems: https://github.com/Toblerity/Shapely/blob/master/.travis.yml. It starts to test at GEOS 3.5).

So how did you install Shapely? Using pip / wheel, or did you install GEOS / Shapely yourself from source?

Also, can you show the full output of running that code? When does it crash? (when reading the file, or when showing `head()`, ..?)

**Comment by antonindanalet** [comment]
I have installed Shapely using Wheel, more specifically using Shapely-1.7.1-cp36-cp36m-win_amd64.whl, if my documentation is correct. It can also be that I have installed it from pip, I am not sure. But for sure **not** from source.

The bug comes from the `read_file`. Maybe some extra info from debugging, if it can help:

```
File "D:/PycharmProjects/my_folder/my_file.py", line 5, in read_function
    world = geopandas.read_file(geopandas.datasets.get_path('naturalearth_lowres'))
File "C:\my_venv\lib\site-packages\geopandas\io\file.py", line 140, in _read_file
    f_filt, crs=crs, columns=columns + ["geometry"]
File "C:\my_venv\lib\site-packages\geopandas\geodataframe.py", line 427, in from_features
  "geometry": shape(feature["geometry"]) if feature["geometry"] else None
File "C:\my_venv\lib\site-packages\shapely\geometry\geo.py", line 111, in shape
   return MultiPolygon(ob["coordinates"], context_type='geojson')
File "C:\my_venv\lib\site-packages\shapely\geometry\multipolygon.py", line 64, in __init__
  self._geom, self._ndim = geos_multipolygon_from_py(polygons)
File "C:\my_venv\lib\site-packages\shapely\geometry\multipolygon.py", line 143, in geos_multipolygon_from_py
  geom, ndims = polygon.geos_polygon_from_py(ob[l][0], ob[l][1:])
File "C:\my_venv\lib\site-packages\shapely\geometry\polygon.py", line 509, in geos_polygon_from_py
  ret = geos_linearring_from_py(shell)
File "shapely\speedups\_speedups.pyx", line 408, in shapely.speedups._speedups.geos_linearring_from_py
ValueError: GEOSGeom_createLinearRing_r returned a NULL pointer
```

Running the script normally does not generate any output, apart from the UserWarning and the exit code.

**Comment by jorisvandenbossche** [comment]
Hmm, it seems to already fail before pygeos is actually involved. Can you try:

```
import geopandas
geopandas.options.use_pygeos = False
world = geopandas.read_file(geopandas.datasets.get_path('naturalearth_lowres'))
```

in the environment *with* pygeos installed?

**Comment by jorisvandenbossche** [comment]
> I have installed Shapely using Wheel, more specifically using Shapely-1.7.1-cp36-cp36m-win_amd64.whl, if my documentation is correct. It can also be that I have installed it from pip, I am not sure. But for sure not from source.

According to https://github.com/shapely/shapely-wheels/blob/master/appveyor.yml, the windows wheels should be built against (and include) GEOS 3.8

Can you run `from shapely.geos import geos_version_string; print(geos_version_string)` to see what that returns?

**Comment by antonindanalet** [comment]
I had already tried `geopandas.options.use_pygeos = False` :) The output is the same in the venv with pygeos: UserWarning + exit code -1073741819 (0xC0000005) + exact same debugging message.

`print(geos_version_string)` returns `3.4.3-CAPI-1.8.3 r4285`.

**Comment by jorisvandenbossche** [comment]
So then it might actually be a problem with the Shapely installation, and not with the use of pygeos or not. The Shapely speedup cython module is built against GEOS 3.8, but for some reason it is finding GEOS 3.4 in your environment. 

Two things to try. Can you see if disabling shapely's speed-ups solve it?

```
import shapely
shapely.speedups.disable()
```

(and then running the code again)

Alternatively, a pure-Shapely reproducer migth be something like

```
from shapely.geometry import Polygon
Polygon([[1, 0], [1, 1], [0, 1], [0, 0], [1, 0]])
```

**Comment by antonindanalet** [comment]
Woohoo, it works when adding ``shapely.speedups.disable()`` with the minimal example! It also works with my larger project, and really really faster 👍 

Thank you very much! Great and super fast answer.

**Comment by jorisvandenbossche** [comment]
@antonindanalet Glad I could help!

**Comment by greatvovan** [comment]
> Woohoo, it works when adding `shapely.speedups.disable()` with the minimal example! It also works with my larger project, and really really faster 👍

Hm... I wonder, disabling these speedups makes it work faster? Do you understand?

**Comment by jorisvandenbossche** [comment]
It was only specific to @antonindanalet's case, because the `speedups` module of shapely failed due to an installation issue. So only by disabling it, the code would run. But it is certainly not a generic solution (also, if using pygeos, enabling or disabling this aspect of Shapely should not have any impact)

**Comment by rodreras** [comment]
> shapely.speedups.disable()

I don't know why, but this worked for me as well. I faced a different error:
`ValueError: GEOSGeom_createLinearRing_r returned a NULL pointer`

The wierd thing is that last time I ran my code, this error didn't show up. 

I installed geopandas, gdal, shapely, fiona, pyproj manually from [this website](https://www.lfd.uci.edu/~gohlke/pythonlibs/). And installed `pygeos` from `pip`.
